### ISSUE TITLE
Does fit reset the model state at the begining of each epoch?

### ISSUE THREAD

**Issue by nickgeoca** [fchollet/keras#2849]
This is relevant for recurrent networks where stateful is True. And when training the model and/or logging the loss.
```python
        model.fit( data['x_train']
                 , data['t_train']
                 , validation_data = (data['x_validation'], data['y_validation'])
                 , batch_size = batchSize
                 , nb_epoch = epochs
                 , verbose = 0
                 , shuffle = False)
```

**Comment by sadeghmir** [comment]
The stateful example `stateful_lstm.py` uses `model.reset_states()` to reset to do that, so no.
